### PR TITLE
feat: Use branchPrefix while onboarding

### LINKED ISSUE #3443: --branch-prefix is ignored for the onboarding branch
**What Renovate type are you using?**
Renovate CLI - Platform: BitBucket Server

**Describe the bug**
--branch-prefix is ignored for the renovate/configure branch when forcing onboarding PR

**Did you see anything helpful in debug logs?**
When running verbose mode it showed me the error of our bitbucket server that the branch was not according to our rules. In the actual logs it said: `INFO: Repository has changed during renovation - aborting` which was not the correct message.

**To Reproduce**

1.  Create repository with branch configuration which does not allow 'renovate/' as branch name
2.  Configure Renovate
3.  Onboarding PR should not be created

**Expected behavior**
--branch-prefix should be included in the onboarding branch name so it will not fail

### PR DESCRIPTION
**Pull request by rakeshtembhurne**
By default, onboarding creates a branch with the name `renovate/configure`.
With `--branch-prefix` CLI option or with `branchPrefix` configuration,
its onboarding branch can be prefixed with the text passed instead of
default `renovate/` prefix.

Closes: #3443

### CODE REVIEW

**Comment by rarkins** [comment]
@rakeshtembhurne have you tested this against a live repository? e.g.

- Onboarding PR is raised with custom branch
- If onboarding PR is closed unmerged then Renovate doesn't create a new one to replace it

**Comment by rakeshtembhurne** [comment]
> @rakeshtembhurne have you tested this against a live repository? e.g.
> 
> * Onboarding PR is raised with custom branch
Yes, I have tested that the onboarding PR is raised with the custom branch.
> * If onboarding PR is closed unmerged then Renovate doesn't create a new one to replace it
Yes, I tested with closed unmerged branch ass well. It does not create a new one to replace it.

**Review by rarkins — CHANGES_REQUESTED** [review]
Set `config.onboardingBranch` as part of the `checkOnboardingBranch` function, instead of adding the new `getOnboardingBranch` function.

File: lib/workers/repository/onboarding/branch/index.js [hunk 1]
```diff
@@ -29,12 +29,20 @@ async function checkOnboardingBranch(config) {
...
-  const branchList = [onboardingBranch];
+  const branchList = [config.onboardingBranch];
   return { ...config, repoIsOnboarded, branchList };
 }
 
+function getOnboardingBranch(config) {
+  if (config.branchPrefix) {
+    return `${config.branchPrefix}configure`;
+  }
+  return onboardingBranch;
```
[the remark this answers is not in the document]
Comment (rakeshtembhurne) [reply lib/workers/repository/onboarding/branch/index.js]: Okay, will make the changes.

File: lib/workers/repository/init/index.js [hunk 2]
```diff
@@ -19,6 +22,7 @@ async function initRepo(input) {
   config.global = config.global || {};
   config = await initApis(config);
   config.semanticCommits = await detectSemanticCommits(config);
+  config.onboardingBranch = getOnboardingBranch(config);
```
[the remark this answers is not in the document]
Comment (rakeshtembhurne) [reply lib/workers/repository/init/index.js]: It can be done from inside as well. Will make the changes.

[the remark this answers is not in the document]
Comment (rakeshtembhurne) [reply lib/workers/repository/init/index.js]: The changes have been made.

File: lib/workers/repository/onboarding/branch/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rakeshtembhurne) [reply lib/workers/repository/onboarding/branch/index.js]: Changes have been made.

**Comment by rarkins** [comment]
This still needs testing. The first scenario I tried (open an onboarding PR, close it, then run Renovate again) did not work - Renovate recreated the onboarding PR. I have fixed this, but there are still other scenarios to test, e.g. rebasing.

**Comment by rarkins** [comment]
@martyganz are you able to test if this PR behaves as you'd expect?

**Comment by rarkins** [comment]
Someone else will need to resume this later if they have a need for it

**Pull request closed without merging**